src/kubernetes.h

```diff
@@ -216,10 +207,12 @@ class KubernetesUpdater : public PollingMetadataUpdater {
 
   KubernetesReader reader_;
   HealthChecker* health_checker_;
-  std::thread node_watch_thread_;
-  std::thread pod_watch_thread_;
-  std::thread service_watch_thread_;
-  std::thread endpoints_watch_thread_;
+  // WatchId combines the plural kubernetes kind, and API version.
+  using WatchId = std::pair<const char*, const char*>;
```

Comment:
This list is only for object types that we know up front.
The watch thread map is not static, and can be easily updated dynamically. 

Also, the alternative you suggest is complaining about the pair not having a non-trivial destructor. Can you point to some relevant docs about global constants?

Reply:
Apologies, `constexpr` is a red herring and won't work in C++11 — I've updated the proposed example above and verified that it works.

The problem with using `WatchId` as currently defined with a dynamic map is that you won't be able to construct one from a `std::string` value, since it requires a `const char*`.